### PR TITLE
Add Annotations

### PR DESCRIPTION
**Pull request by philippfromme**
# Features

* add linting annotations to canvas
* add single API to update both annotations and properties panel errors

image

---

Related to https://github.com/bpmn-io/internal-docs/issues/591

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,12 +8,24 @@ The BPMN linter used by the Camunda Desktop and Web Modeler. Batteries included.
...
 
 # Usage
 
 ```javascript
+import Modeler from 'bpmn-js/lib/Modeler';
+
 import { Linter } from '@camunda/linting';
-import { getErrors } from '@camunda/linting/properties-panel';
+
+import lintingModule from '@camunda/linting/modeler';
```
Comment (pinussilvestrus) [inline README.md:20]: Will we still include that in the Camunda Modeler or now rather in `camunda-bpmn-js`?

File: README.md [hunk 2]
```diff
@@ -8,12 +8,24 @@ The BPMN linter used by the Camunda Desktop and Web Modeler. Batteries included.
...
+
+import '@camunda/linting/assets/linting.css';
```
Comment (pinussilvestrus) [inline README.md:22]: Just fyi, in the desktop modeler will need to add this one to our [global modeling styles](https://github.com/camunda/camunda-modeler/blob/develop/client/src/styles/_modeling.less).

File: README.md (hunk 1 above)
Comment (philippfromme) [reply README.md:20]: This will have to be integrated into the Desktop and Web Modeler. It's not part of `camunda-bpmn-js`.

File: README.md (hunk 2 above)
Comment (philippfromme) [reply README.md:22]: 👍🏻

File: lib/modeler/LintingAnnotations.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { groupBy } from 'min-dash';
+
+import { domify } from 'min-dom';
```
Comment (pinussilvestrus) [inline lib/modeler/LintingAnnotations.js:3]: This is not a dependency right now.

File: test/spec/Linter.spec.js [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+import camundaCloud10InvalidXML from  './camunda-cloud-1-0-invalid.bpmn';
+import camundaCloud11ValidXML from  './camunda-cloud-1-1-valid.bpmn';
+import camundaCloud11InvalidXML from  './camunda-cloud-1-1-invalid.bpmn';
+import camundaCloud12ValidXML from  './camunda-cloud-1-2-valid.bpmn';
+import camundaCloud12InvalidXML from  './camunda-cloud-1-2-invalid.bpmn';
+import camundaCloud13ValidXML from  './camunda-cloud-1-3-valid.bpmn';
+import camundaCloud13InvalidXML from  './camunda-cloud-1-3-invalid.bpmn';
+import camundaCloud80ValidXML from  './camunda-cloud-8-0-valid.bpmn';
+import camundaCloud80InvalidXML from  './camunda-cloud-8-0-invalid.bpmn';
+import camundaPlatform717XML from  './camunda-platform-7-17.bpmn';
```
Comment (pinussilvestrus) [inline test/spec/Linter.spec.js:18]:
```suggestion
import noExecutionPlatformXML from './no-execution-platform.bpmn';
import camundaCloud10ValidXML from './camunda-cloud-1-0-valid.bpmn';
import camundaCloud10InvalidXML from './camunda-cloud-1-0-invalid.bpmn';
import camundaCloud11ValidXML from './camunda-cloud-1-1-valid.bpmn';
import camundaCloud11InvalidXML from './camunda-cloud-1-1-invalid.bpmn';
import camundaCloud12ValidXML from './camunda-cloud-1-2-valid.bpmn';
import camundaCloud12InvalidXML from './camunda-cloud-1-2-invalid.bpmn';
import camundaCloud13ValidXML from './camunda-cloud-1-3-valid.bpmn';
import camundaCloud13InvalidXML from './camunda-cloud-1-3-invalid.bpmn';
import camundaCloud80ValidXML from './camunda-cloud-8-0-valid.bpmn';
import camundaCloud80InvalidXML from './camunda-cloud-8-0-invalid.bpmn';
import camundaPlatform717XML from './camunda-platform-7-17.bpmn';
```

Comment (pinussilvestrus) [reply test/spec/Linter.spec.js:18]: Too many spaces 🤡

**Review by pinussilvestrus — COMMENTED** [review]
I ❤️ that we now have the linting in an own bpmn-js extension! Code-wise it looks good to me. 

Usability also looks good- as I am not super familiar with what was decided, I'd wait for a 👍 from someone else (@CatalinaMoisuc, @christian-konrad).

File: test/spec/Linter.spec.js (hunk 4 above)
Comment (philippfromme) [reply test/spec/Linter.spec.js:18]: 👀

File: lib/modeler/LintingAnnotations.js (hunk 3 above)
Comment (philippfromme) [reply lib/modeler/LintingAnnotations.js:3]: 👀

Comment (philippfromme) [reply lib/modeler/LintingAnnotations.js:3]: Added: https://github.com/camunda/linting/pull/11/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R31

File: test/spec/Linter.spec.js (hunk 4 above)
Comment (philippfromme) [reply test/spec/Linter.spec.js]: Fixed: https://github.com/camunda/linting/pull/11/files#diff-2ae0792d4da08f8dd39cf81edb9d3f2c1ec21d54f0db70d72c569af6b203b1adR7

**Review by pinussilvestrus — APPROVED** [review]
👍 from my side

**Pull request merged**